File: bullhornatsclone_ui/src/services/MatchService.js
import http from "../http-common"; 

class MatchService {
  getAllMatchs(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/match/matchs`, searchDTO);
  }

  get(matchId) {
    return this.getRequest(`/match/${matchId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/match?field=${matchData}`, null);
  }

  addMatch(data) {
    return http.post("/match/addMatch", data);
  }

  update(data) {
  	return http.post("/match/updateMatch", data);
  }
  
  uploadImage(data,matchId) {
  	return http.postForm("/match/uploadImage/"+matchId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new MatchService();

File: src/main/java/com/bullhornatsclone/service/MatchService.java
package com.bullhornatsclone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.bullhornatsclone.domain.Match;
import com.bullhornatsclone.dto.MatchDTO;
import com.bullhornatsclone.dto.MatchSearchDTO;
import com.bullhornatsclone.dto.MatchPageDTO;
import com.bullhornatsclone.dto.MatchConvertCriteriaDTO;
import com.bullhornatsclone.service.GenericService;
import com.bullhornatsclone.dto.common.RequestDTO;
import com.bullhornatsclone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface MatchService extends GenericService<Match, Integer> {

	List<Match> findAll();

	ResultDTO addMatch(MatchDTO matchDTO, RequestDTO requestDTO);

	ResultDTO updateMatch(MatchDTO matchDTO, RequestDTO requestDTO);

    Page<Match> getAllMatchs(Pageable pageable);

    Page<Match> getAllMatchs(Specification<Match> spec, Pageable pageable);

	ResponseEntity<MatchPageDTO> getMatchs(MatchSearchDTO matchSearchDTO);
	
	List<MatchDTO> convertMatchsToMatchDTOs(List<Match> matchs, MatchConvertCriteriaDTO convertCriteria);

	MatchDTO getMatchDTOById(Integer matchId);







}





